spec/models/assignment_spec.rb

```diff
@@ -1808,4 +1808,123 @@ def grouping_count(groupings)
       end
     end
   end
+
+  describe '#summary_json' do
+    context 'a Student user' do
+      let(:assignment) { create :assignment }
+      let(:student) { create :student }
+
+      it 'should return {}' do
+        expect(assignment.summary_json(student)).to be_empty
+      end
+    end
+
+    context 'a TA user' do
+      let(:ta) { create :ta}
+
+      before :each do
+        @assignment = create(:assignment_with_criteria_and_results)
+      end
+
+      context 'with no assigned students' do
+        it 'has criteria columns' do
+          expect(@assignment.summary_json(ta)[:criteriaColumns]).not_to be_empty
+        end
+
+        it 'has correct criteria information' do
+          criteria_info = @assignment.summary_json(ta)[:criteriaColumns][0]
+          expect(criteria_info).is_a? Hash
+          expect(criteria_info.key? :Header)
+          expect(criteria_info.key? :accessor)
+          expect(criteria_info.key? :className)
+        end
+      end
+    end
+
+    context 'an Admin user' do
+      let(:admin) { create :admin }
+
+      before :each do
+        @assignment = create(:assignment_with_criteria_and_results)
+      end
+
+      context 'with assigned students' do
+        it 'has criteria columns' do
+          expect(@assignment.summary_json(admin)[:criteriaColumns]).not_to be_empty
+        end
+
+        it 'has correct criteria information' do
+          criteria_info = @assignment.summary_json(admin)[:criteriaColumns][0]
+          expect(criteria_info).is_a? Hash
+          expect(criteria_info.key? :Header)
```

Comment:
Instead of checking for the existence of each key one by one, we should be checking for them all at once. That way, if we are missing multiple keys, our test will report that they are all missing at the same time. 

Have a look at rspec's [contain_exactly](https://relishapp.com/rspec/rspec-expectations/v/3-9/docs/built-in-matchers/contain-exactly-matcher) matcher